src/webgpu/shader/validation/shader_validation_test.ts

```diff
@@ -0,0 +1,42 @@
+import { GPUTest } from '../../gpu_test.js';
+
+export class ShaderValidationTest extends GPUTest {
+  /**
+   * Add a test expectation for whether a createShaderModule call succeeds or not.
+   *
+   * @example
+   * ```ts
+   * t.expectCompileResult(true, `wgsl code`); // Expect success
+   * t.expectCompileResult(false, `wgsl code`); // Expect validation error with any error string
+   * t.expectCompileResult('v-0000', `wgsl code`); // Expect validation error containing 'v-0000'
+   * ```
+   */
+  expectCompileResult(result: boolean | string, code: string) {
+    // If no result is expected, we let the scope surrounding the test catch it.
```

Comment:
nit: perhaps this comment should be reversed: "if an error is expected, push a scope.."